pkg/arvo/app/pool-group-hook.hoon

```diff
@@ -0,0 +1,199 @@
+  ::
+  ++  on-init
+    ^-  (quip card _this)
+    =^  cards  state
+      poll-inviter:do
+    [cards this]
+  ::
+  ++  on-save   !>(state)
+  ++  on-load
+    |=  old=vase
+    ^-  (quip card _this)
+    [~ this(state !<(app-state old))]
+  ::
+  ++  on-agent
+    |=  [=wire =sign:agent:gall]
+    ^-  (quip card agent:gall)
+    ?.  ?=([%running *] wire)
+      (on-agent:def wire sign)
+    ?-  -.sign
+        %poke-ack
+      ?~  p.sign
+        [~ this]
+      %-  (slog leaf+"{(trip dap.bowl)} couldn't start thread" u.p.sign)
+      :_  this(running.state ~)
+      ~[(leave-spider:do t.wire) set-timer:do]
+    ::
+        %watch-ack
+      ?~  p.sign
+        [~ this]
+      =/  =tank  leaf+"{(trip dap.bowl)} couldn't start listen to thread"
+      %-  (slog tank u.p.sign)
+      [[set-timer:do]~ this(running.state ~)]
+    ::
+        %kick  [~ this(running.state ~)]
+        %fact
+      ?+  p.cage.sign  (on-agent:def wire sign)
+          %thread-fail
+        =+  !<([=term =tang] q.cage.sign)
+        %-  (slog leaf+"{(trip dap.bowl)} failed; will retry" leaf+<term> tang)
+        [[set-timer:do]~ this(running.state ~)]
+      ::
+          %thread-done
+        ?+  t.wire  ~|([dap.bowl %unexpected-thread-done wire] !!)
+            [%inviter ~]
+          =+  !<(res=@t q.cage.sign)
+          =/  inviter=ship
+            `@`(decode-results:abi:ethereum res [%uint]~)
+          ::  if we weren't invited by anyone, don't do anything anymore.
+          ::
+          ?:  =(0 inviter)  [~ this(state state(running ~, inviter ~))]
+          =.  inviter.state  `inviter
+          =^  cards  state
+            poll-invited:do
+          [cards this]
+        ::
+            [%invited ~]
+          =+  !<(res=@t q.cage.sign)
+          =/  invited=(list ship)
+            ;;  (list ship)
+            (decode-results:abi:ethereum res [%array %uint]~)
+          =^  cards  state
+            (process-invited:do invited)
+          [cards this(running.state ~)]
+        ==
+      ==
+    ==
+  ::
+  ++  on-arvo
+    |=  [=wire =sign-arvo]
+    ^-  (quip card _this)
+    ?+  +<.sign-arvo  ~|([dap.bowl %strange-arvo-sign +<.sign-arvo] !!)
+        %wake
+      =^  cards  state
+        ?~  inviter.state
+          poll-inviter:do
+        poll-invited:do
+      [cards this]
+    ==
+  ::
+  ++  on-poke   on-poke:def
+  ++  on-peek   on-peek:def
+  ++  on-watch  on-watch:def
+  ++  on-leave  on-leave:def
+  ++  on-fail   on-fail:def
+  --
+::
+|_  =bowl:gall
+++  five   5
+++  poke-spider
+  |=  [=path =cage]
+  ^-  card
+  [%pass [%running path] %agent [our.bowl %spider] %poke cage]
+::
+++  watch-spider
+  |=  [=path =sub=path]
+  ^-  card
+  [%pass [%running path] %agent [our.bowl %spider] %watch sub-path]
+::
+++  leave-spider
+  |=  [=path]
+  ^-  card
+  [%pass [%running path] %agent [our.bowl %spider] %leave ~]
+::
+++  new-tid
+  |=  eny=@uv
+  ^-  @t
+  :((cury cat 3) dap.bowl '--' (scot %uv eny))
```

Comment:
I find it convenient for the `tid` to be a something that gets printed as a literal that can be typed in, for example for use with `+spider/tree` and `:spider|stop`.  If you `(scot %ta)` this then it'll work.  Also I've been using `_` instead of `--`, but that's less important.

Reply:
Ah, that's all fair. Fix incoming!